### ISSUE TITLE
[BUG] build failed in databricks 10.4 runtime (updated recently)

### ISSUE THREAD

**Issue by pxLi** [NVIDIA/spark-rapids#5154]
**Describe the bug**
image of databricks 10.4 runtime seems get updated today.

which failed find some default jar dependencies.
```
[2022-04-07T01:11:37.173Z] [INFO] --- maven-install-plugin:2.4:install-file (default-cli) @ rapids-4-spark-parent ---

[2022-04-07T01:11:37.428Z] [ERROR] The specified file '/databricks/jars/----workspace_spark_3_2--maven-trees--hive-2.3__hadoop-3.2--org.apache.parquet--parquet-hadoop--org.apache.parquet__parquet-hadoop__1.12.0-databricks-0003.jar' not exists
```
![image](https://user-images.githubusercontent.com/8086184/162120858-28398f3b-c975-4a97-bdf5-c4efe246c2d3.png)

tried (https://github.com/NVIDIA/spark-rapids/pull/5153) bump up some versions, and then it failed due to some func signature changes,
```
[2022-04-07T04:19:32.450Z] [INFO] Compiling 273 Scala sources and 29 Java sources to /home/ubuntu/spark-rapids/sql-plugin/target/spark321db/classes...
[2022-04-07T04:19:34.954Z] [ERROR] /home/ubuntu/spark-rapids/sql-plugin/src/main/311+-db/scala/com/nvidia/spark/rapids/shims/AQEUtils.scala:28: not enough arguments for method apply: (id: Int, plan: org.apache.spark.sql.execution.SparkPlan, originalPlan: org.apache.spark.sql.execution.SparkPlan, isSparkExchange: Boolean)org.apache.spark.sql.execution.adaptive.ShuffleQueryStageExec in object ShuffleQueryStageExec.
[2022-04-07T04:19:34.955Z] Unspecified value parameter isSparkExchange.
[2022-04-07T04:19:34.955Z] [ERROR]     ShuffleQueryStageExec(sqse.id, reusedExchange, sqse.originalPlan)
[2022-04-07T04:19:34.955Z] [ERROR]                          ^
[2022-04-07T04:19:35.209Z] [ERROR] /home/ubuntu/spark-rapids/sql-plugin/src/main/311until330-all/scala/com/nvidia/spark/rapids/shims/GpuRangePartitioning.scala:68: wrong number of arguments for pattern org.apache.spark.sql.catalyst.plans.physical.ClusteredDistribution(clustering: Seq[org.apache.spark.sql.catalyst.expressions.Expression],requireAllClusterKeys: Boolean,requiredNumPartitions: Option[Int])
[2022-04-07T04:19:35.209Z] [ERROR]         case ClusteredDistribution(requiredClustering, _) =>
[2022-04-07T04:19:35.209Z] [ERROR]                                   ^
[2022-04-07T04:19:35.766Z] [ERROR] /home/ubuntu/spark-rapids/sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/Spark320PlusShims.scala:926: wrong number of arguments for pattern org.apache.spark.sql.execution.adaptive.ShuffleQueryStageExec(id: Int,plan: org.apache.spark.sql.execution.SparkPlan,originalPlan: org.apache.spark.sql.execution.SparkPlan,isSparkExchange: Boolean)
[2022-04-07T04:19:35.766Z] [ERROR]     case ShuffleQueryStageExec(_, e: ReusedExchangeExec, _) => e
[2022-04-07T04:19:35.766Z] [ERROR]                               ^
[2022-04-07T04:19:35.766Z] [ERROR] /home/ubuntu/spark-rapids/sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/Spark320PlusShims.scala:927: wrong number of arguments for pattern org.apache.spark.sql.execution.adaptive.BroadcastQueryStageExec(id: Int,plan: org.apache.spark.sql.execution.SparkPlan,originalPlan: org.apache.spark.sql.execution.SparkPlan,isSparkExchange: Boolean)
[2022-04-07T04:19:35.766Z] [ERROR]     case BroadcastQueryStageExec(_, e: ReusedExchangeExec, _) => e
[2022-04-07T04:19:35.766Z] [ERROR]                                 ^
[2022-04-07T04:19:35.766Z] [ERROR] /home/ubuntu/spark-rapids/sql-plugin/src/main/321db/scala/com/nvidia/spark/rapids/shims/GpuHashPartitioning.scala:34: wrong number of arguments for pattern org.apache.spark.sql.catalyst.plans.physical.ClusteredDistribution(clustering: Seq[org.apache.spark.sql.catalyst.expressions.Expression],requireAllClusterKeys: Boolean,requiredNumPartitions: Option[Int])
[2022-04-07T04:19:36.020Z] [ERROR]         case ClusteredDistribution(requiredClustering, _) =>
[2022-04-07T04:19:36.020Z] [ERROR]                                   ^
[2022-04-07T04:19:36.021Z] [ERROR] /home/ubuntu/spark-rapids/sql-plugin/src/main/321db/scala/com/nvidia/spark/rapids/shims/SparkShims.scala:68: not enough arguments for method apply: (id: Int, plan: org.apache.spark.sql.execution.SparkPlan, originalPlan: org.apache.spark.sql.execution.SparkPlan, isSparkExchange: Boolean)org.apache.spark.sql.execution.adaptive.BroadcastQueryStageExec in object BroadcastQueryStageExec.
[2022-04-07T04:19:36.021Z] Unspecified value parameter isSparkExchange.
[2022-04-07T04:19:36.021Z] [ERROR]     BroadcastQueryStageExec(old.id, newPlan, old.originalPlan)
[2022-04-07T04:19:36.021Z] [ERROR]                            ^
[2022-04-07T04:19:48.174Z] [ERROR] 6 errors found
```

**Comment by tgravescs** [comment]
Looks like 2 different changes.  

1) pulled in https://issues.apache.org/jira/browse/SPARK-38237 which was changed in spark 3.3 back
2) change in parameters to ShuffleQueryStageExec which looks new

**Comment by tgravescs** [comment]
Issue above is merged into 22.04 and 22.06, closing.
